### PR TITLE
[Models] Add models (members, courses, scholarships, colleges)

### PR DESCRIPTION
**Pull request by wonglkd**
- Added models and migrations.

Trello: https://trello.com/c/v6FsTC5t/3-models-based-on-existing-schema

### CODE REVIEW

File: app/College.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     *
+     * @var array
+     */
+    protected $fillable = array(
+        'name',
+    );
+
+    public function members()
+    {
+        return $this->belongstoMany('App/members');
```
Comment (spencerthang) [inline app/College.php]: Isn't this belongstoMany('App\Member') - similarly for course and scholarship?

**Comment by wonglkd** [comment]
@spencerthang clearly, you should be the one reviewing this and merging this, tyvm :P responding to comments immediately after

File: database/migrations/2016_03_13_203345_create_members_table.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            $table->integer('end_year');
+
+            $table->datetime('registration_time');
+            $table->enum('membership_type', ['Life', '1 year']);
+            $table->boolean('is_fee_paid');
+            $table->boolean('is_card_issued');
+            $table->text('remarks');
+
+            $table->timestamps();
+        });
```
Comment (spencerthang) [inline database/migrations/2016_03_13_203345_create_members_table.php:49]: Should we make members soft delete?

**Comment by wonglkd** [comment]
@spencerthang yup you're right. thanks for catching this!

File: database/migrations/2016_03_13_203345_create_members_table.php (hunk 2 above)
Comment (wonglkd) [inline database/migrations/2016_03_13_203345_create_members_table.php:49]: Yes, we should!

**Comment by spencerthang** [comment]
@wonglkd Laravel has a default User class which provides for authentication, i.e. it rolls in built in auth, register, forget pass, etc. with this class.

I can see User having a 1:1 relationship with Member, or just extending User with what you intend to chuck into Member. I prefer the latter slightly, because otherwise you'll always have to do Auth::user()->member->dob for example, instead of just Auth::user()->dob.

File: database/migrations/2016_03_13_203345_create_members_table.php (hunk 2 above)
Comment (wonglkd) [inline database/migrations/2016_03_13_203345_create_members_table.php:49]: @spencerthang soft deletes for every table?

**Comment by wonglkd** [comment]
@spencerthang Yes, the plan was to have User be a 1:1 relationship with Member, I think, since you could have a Member without a User (say, if they have never logged in before). What do you think, hmm?

**Comment by spencerthang** [comment]
> since you could have a Member without a User (say, if they have never logged in before)

@wonglkd How would this work if a member wanted to sign up in future? Do they create a new user and claim a member?

I was envisioning members having their emails be their crsid emails by default, and them using forget password (if they didn't sign up via the site, but rather through fresher's fair or something)... or Raven integration if we ever get around to it.

**Comment by wonglkd** [comment]
> ```
> since you could have a Member without a User (say, if they have never logged in before)
> ```
> 
> @wonglkd How would this work if a member wanted to sign up in future? Do they create a new user and claim a member?
> 
> I was envisioning members having their emails be their crsid emails by default, and them using forget password (if they didn't sign up via the site, but rather through fresher's fair or something)... or Raven integration if we ever get around to it.

Yup! They create a new user and claim a member. If their CRSid is present, they can auto-claim, otherwise it can be trigger an email to DBO and be manually associated? Or something like that? According to @tpl2go and Darren, users who do not have CRSids are a real thing that happens... In light of that what do you think?

**Comment by spencerthang** [comment]
@wonglkd Ah that's annoying, I do recall that not all CUMSA members are University members. Yep that's fine by me!

File: database/migrations/2016_03_13_203345_create_members_table.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            $table->string('first_name');
+            $table->enum('gender', ['Male', 'Female']);
+
+            // Nationality and DOB are nullable because some people do not want to declare.
+            $table->date('date_of_birth')->nullable();
+            $table->string('nationality')->nullable();
+
+            $table->boolean('is_singapore_pr');
+            $table->string('nric');
+            $table->string('crsid')->unique();
```
Comment (tpl2go) [inline database/migrations/2016_03_13_203345_create_members_table.php:28]: we should make crsid nullable. Some non university members might join cumsa @spencerthang

File: database/migrations/2016_03_13_203345_create_members_table.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+            $table->string('email_hermes');
```
Comment (tpl2go) [inline database/migrations/2016_03_13_203345_create_members_table.php:30]: some department emails are not derivable from crsid. hence this column

**Pull request merged**